govcd/lbservicemonitor_test.go

```diff
@@ -0,0 +1,79 @@
+// +build lb lbServiceMonitor functional ALL
+
+/*
+ * Copyright 2019 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	. "gopkg.in/check.v1"
+)
+
+// Test_LBServiceMonitor tests CRUD methods for load balancer service monitor.
+// The following things are tested:
+// Creation of load balancer service monitor
+// Read load balancer by both ID and Name (service monitor name must be unique in single edge gateway)
+// Update - change a single field and compare that configuration and result objects are deeply equal
+// Update - try and fail to update without mandatory field
+// Delete
+func (vcd *TestVCD) Test_LBServiceMonitor(check *C) {
+	if vcd.config.VCD.EdgeGateway == "" {
+		check.Skip("Skipping test because no edge gateway given")
+	}
+	edge, err := vcd.vdc.FindEdgeGateway(vcd.config.VCD.EdgeGateway)
+	check.Assert(err, IsNil)
+	check.Assert(edge.EdgeGateway.Name, Equals, vcd.config.VCD.EdgeGateway)
+
```

Comment:
Before running operations, we should check that the edge gateway has Advanced Networking enabled.
If it is not, we will get a nasty message.
```
lbservicemonitor_test.go:39:
    check.Assert(err, IsNil)
... value *errors.errorString = &errors.errorString{s:"error creating load balancer service monitor:  
This operation is denied. (API error: 403)"} ("error creating load balancer service monitor:  This operation is denied. (API error: 403)")
```

Reply:
Should be better now with additional check.